### PR TITLE
feat(kms): support to config key rotation

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- huaweicloud_kms_key: support to config key rotation;
- data.huaweicloud_kms_key: Add attributes of key rotation;

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud' TESTARGS='-run TestAccKmsKey_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccKmsKey_Basic -timeout 360m -parallel 4
=== RUN   TestAccKmsKey_Basic
=== PAUSE TestAccKmsKey_Basic
=== CONT  TestAccKmsKey_Basic
--- PASS: TestAccKmsKey_Basic (71.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       71.925s

$ make testacc TEST='./huaweicloud' TESTARGS='-run TestAccKmsKey_rotation'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccKmsKey_rotation -timeout 360m -parallel 4
=== RUN   TestAccKmsKey_rotation
=== PAUSE TestAccKmsKey_rotation
=== CONT  TestAccKmsKey_rotation
--- PASS: TestAccKmsKey_rotation (99.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       99.579s

$ make testacc TEST='./huaweicloud' TESTARGS='-run TestAccKmsKeyDataSource_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccKmsKeyDataSource_Basic -timeout 360m -parallel 4
=== RUN   TestAccKmsKeyDataSource_Basic
=== PAUSE TestAccKmsKeyDataSource_Basic
=== CONT  TestAccKmsKeyDataSource_Basic
--- PASS: TestAccKmsKeyDataSource_Basic (37.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       37.255s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_kms_key_v1.go [hunk 1]
```diff
@@ -68,11 +70,23 @@ func ResourceKmsKeyV1() *schema.Resource {
...
 			},
+			"rotation_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"rotation_interval": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				RequiredWith: []string{"rotation_enabled"},
```
Comment (hmbig2) [inline huaweicloud/resource_huaweicloud_kms_key_v1.go:81]: move RequiredWith to `rotation_enabled`, and remove resourceKmsKeyValidation

File: huaweicloud/resource_huaweicloud_kms_key_v1.go [hunk 2]
```diff
@@ -269,9 +341,37 @@ func resourceKmsKeyV1Update(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	if d.HasChange("tags") {
-		tagErr := utils.UpdateResourceTags(kmsKeyV1Client, d, "kms", d.Id())
+		tagErr := utils.UpdateResourceTags(kmsKeyV1Client, d, "kms", keyID)
 		if tagErr != nil {
-			return fmtp.Errorf("Error updating tags of kms:%s, err:%s", d.Id(), err)
+			return fmtp.Errorf("Error updating tags of kms:%s, err:%s", keyID, err)
```
Comment (hmbig2) [inline huaweicloud/resource_huaweicloud_kms_key_v1.go]: s/Error/error

File: huaweicloud/resource_huaweicloud_kms_key_v1.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/resource_huaweicloud_kms_key_v1.go:81]: the `rotation_interval` can be optional when rotation is enabled (defaults to 365)
so we must using **RequiredWith** in rotation_interval

File: huaweicloud/resource_huaweicloud_kms_key_v1.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/resource_huaweicloud_kms_key_v1.go]: fixed

**Pull request merged**